### ISSUE TITLE
backward of stack is wrong when sbp  is s1

### ISSUE THREAD

**Issue by liufengwei0103** [Oneflow-Inc/oneflow#8194]
```
import numpy as np
import oneflow as flow
placement = flow.placement("cuda", ranks=[0, 1])
sbp = flow.sbp.split(1)
a = flow.randn(2, 2, placement=placement, sbp=sbp, dtype=flow.float, requires_grad=True)
b = flow.randn(2, 2, placement=placement, sbp=sbp, dtype=flow.float, requires_grad=True)

np_arr = np.arange(8).reshape(2,2,2)
diff = flow.tensor(np_arr, placement=placement, sbp=flow.sbp.broadcast, dtype=flow.float)
c = flow.stack((a, b), dim=0)
c.backward(diff)
print(a.grad)
```

I got:
tensor([[0., 2.],
        [1., 3.]], placement=oneflow.placement(type="cuda", ranks=[0, 1]),
       sbp=(oneflow.sbp.split(axis=1),), dtype=oneflow.float32)
 
the right one is 
tensor([[0., 1.],
        [2., 3.]], placement=oneflow.placement(type="cuda", ranks=[0, 1]),
       sbp=(oneflow.sbp.split(axis=1),), dtype=oneflow.float32)

**Comment by MARD1NO** [comment]
will be fixed in https://github.com/Oneflow-Inc/oneflow/pull/8181

<img width="948" alt="image" src="https://user-images.githubusercontent.com/42901638/168251107-48726e5b-b9ba-45b7-b5b2-146db95c4267.png">

**Comment by MARD1NO** [comment]
https://github.com/Oneflow-Inc/oneflow/pull/8181 has been merged
